tests/rados/test_mon_system_operations.py

```diff
@@ -0,0 +1,272 @@
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Module to test Mon daemons on RHCS clusters
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonitorWorkflows(node=cephadm)
+    cluster_nodes = ceph_cluster.get_nodes()
+    init_mon_nodes = ceph_cluster.get_nodes(role="mon")
+
+    pool_name = "mon_service_test_pool"
+    if not rados_obj.create_pool(pool_name=pool_name):
+        log.error(f"Could not create pool {pool_name} for mon tests")
+        raise Exception("Pool not created error")
+
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    log.info("---------- starting workflow 1 - Addition of new mons --------")
+    # Setting the mon service as unmanaged by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    found = False
+    # Idenitfying test host without any mons to deploy new mons
+    for item in cluster_nodes:
+        if item.hostname not in [mon.hostname for mon in init_mon_nodes]:
+            alt_mon_host = item
+            found = True
+            break
+    if not found:
+        log.error("No alternate mon host was found... fail")
+        raise Exception("No free mon host found error")
+
+    log.debug(
+        f"Selected host: {alt_mon_host.hostname} with "
+        f"IP : {alt_mon_host.ip_address} to deploy new mon daemon"
+    )
+
+    # Adding the new mon to the cluster
+    if not mon_obj.add_mon_service(host=alt_mon_host):
+        log.error(f"Could not add mon service on host {alt_mon_host.hostname} ")
+        raise Exception("mon service not added error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    # Checking if the new mon added is part of Mon Quorum
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if alt_mon_host.hostname not in quorum:
+        log.error(
+            f"selected host : {alt_mon_host.hostname} does not have mon as part of Quorum post addition"
+        )
+        raise Exception("Mon not in quorum error")
+
+    log.info("Completed workflow 1. Addition of new mon daemon")
+
+    log.info("---------- starting workflow 2 - Removal of new mons --------")
+    # Setting the mon service as unmanaged
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    # Removing mon service
```

Comment:
Please perform these checks to see if removing the mon would break the cluster health
`mon ok-to-rm <id>      check whether removing the specified mon would break quorum`
`mon ok-to-stop <ids>...    check whether mon(s) can be safely stopped without reducing immediate availability`

Reply:
I will take this up as enhancement in next PR.